### PR TITLE
[PM-10325] Rename OrganizationUser Delete and BulkDelete endpoints to Remove and BulkRemove

### PR DESCRIPTION
**Pull request by r-tome**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-10325

## 📔 Objective

We're renaming the existing `OrganizationUsersController` `Delete` and `BulkDelete` endpoints to `Remove` and `BulkRemove` to make it clear that we are removing the user from the organization, not deleting their data. Additionally, we're renaming the `IDeleteOrganizationUserCommand` to `IRemoveOrganizationUserCommand`.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

**Review by ike-kottlowski — APPROVED** [review]
Auth changes are good.

File: src/Api/AdminConsole/Public/Controllers/MembersController.cs [hunk 1]
```diff
@@ -243,7 +243,7 @@ public async Task<IActionResult> Delete(Guid id)
         {
             return new NotFoundResult();
         }
-        await _organizationService.DeleteUserAsync(_currentContext.OrganizationId.Value, id, null);
+        await _organizationService.RemoveUserAsync(_currentContext.OrganizationId.Value, id, null);
```
Comment (vincentsalucci) [inline src/Api/AdminConsole/Public/Controllers/MembersController.cs:246]: 💭 Do we have plans to update the Public API endpoint as well?

**Review by vincentsalucci — DISMISSED** [review]
Just a question about our future plans. Everything else looks good 👍

File: src/Api/AdminConsole/Public/Controllers/MembersController.cs (hunk 1 above)
Comment (r-tome) [reply src/Api/AdminConsole/Public/Controllers/MembersController.cs:246]: Great point, yes! I've renamed that endpoint too.

**Review by amorask-bitwarden — APPROVED** [review]
Billing changes look good.

File: src/Api/AdminConsole/Public/Controllers/MembersController.cs (hunk 1 above)
Comment (vincentsalucci) [reply src/Api/AdminConsole/Public/Controllers/MembersController.cs:246]: 💭 This didn't change the endpoint itself (which is fine, just pointing it out), I'm assuming future changes there will require some finesse. 
cc @eliykat

**Review by vincentsalucci — APPROVED** [review]

**Review by ike-kottlowski — APPROVED** [review]

**Pull request merged**